File: qos-mom-client/src/distribution/message/Subscription.java
package distribution.message;


public abstract class Subscription {

	private String topic;

	private int port;
	
	private String ip;

	public abstract void onMessage(String topic, Message msg);

	public String getTopic() {
		return topic;
	}

	public void setTopic(String topic) {
		this.topic = topic;
	}

	public int getPort() {
		return port;
	}

	public void setPort(int port) {
		this.port = port;
	}

	public String getIp() {
		return ip;
	}

	public void setIp(String ip) {
		this.ip = ip;
	}
}

File: ruby-publisher/subscription.rb
require 'json'
require_relative 'message'

class Subscription

	attr_accessor :topic, :ip, :port

	def initialize 
		@topic = ""
		@ip = ""
		@port = 0
	end

	def on_message(topic, msg)
		
	end

	def to_json
		hash = {}
        self.instance_variables.each do |var|
            hash[var.to_s.gsub('@','')] = self.instance_variable_get var
        end
        hash.to_json
	end
end